File: задание 1/main.cs
using System;

class Program
{
    static void Main()
    {
        int N, M;
        Console.Write("Input length: ");
        while (!int.TryParse(Console.ReadLine(), out N))
        {
            Console.WriteLine("Invalid input. Please enter a valid integer for length.");
            Console.Write("Input length: ");
        }

        Console.Write("Input width: ");
        while (!int.TryParse(Console.ReadLine(), out M))
        {
            Console.WriteLine("Invalid input. Please enter a valid integer for width.");
            Console.Write("Input width: ");
        }

        Console.Write("Input all commands: ");
        string commands = Console.ReadLine();

        int x = 0, y = 0;

        foreach (char command in commands)
        {
            switch (command)
            {
                case 'L':
                    x -= 1;
                    break;
                case 'R':
                    x += 1;
                    break;
                case 'D':
                    y += 1;
                    break;
                case 'U':
                    y -= 1;
                    break;
                default:
                    Console.WriteLine($"Invalid command - {command}");
                    return;
            }

            if (Math.Abs(y) >= N || Math.Abs(x) >= M)
            {
                Console.WriteLine("No");
                return;
            }
        }

        Console.WriteLine($"({M - Math.Abs(x)},{N - Math.Abs(y)})");
    }
}

File: задание 1/main.rs
use std::io::{self, Write};

fn main() {
    let mut input = String::new();

    print!("Input length: ");
    io::stdout().flush().unwrap();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let N: i32 = input.trim().parse().expect("Please enter a valid integer for length");

    input.clear();
    print!("Input width: ");
    io::stdout().flush().unwrap();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let M: i32 = input.trim().parse().expect("Please enter a valid integer for width");

    input.clear();
    print!("Input all commands: ");
    io::stdout().flush().unwrap();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let commands = input.trim();

    let mut x: i32 = 0;
    let mut y: i32 = 0;

    for command in commands.chars() {
        match command {
            'L' => x -= 1,
            'R' => x += 1,
            'D' => y += 1,
            'U' => y -= 1,
            _ => {
                println!("Invalid command - {}", command);
                return;
            }
        }

        if y.abs() >= N || x.abs() >= M {
            println!("No");
            return;
        }
    }

    println!("({}, {})", M - x.abs(), N - y.abs());
}
